airflow/api/common/mark_tasks.py

```diff
@@ -79,6 +79,7 @@ def _create_dagruns(
 def set_state(
     *,
     tasks: Iterable[Operator],
+    map_indexes: Optional[Iterable[int]] = None,
```

Comment:
I thought about this a while ago, passing these as two arguments seems wrong because generally it only makes sense if either `tasks` are all non-mapped, or contain only one task. It makes very little sense to do something like

```python
set_state(tasks=[task_1, task_2], map_indexes=[1, 3])
```

Instead, we should probably make `tasks` take 2-tuples like this:

```python
set_state(
    tasks=[
        (task_1, 2),  # The specific mapped ti.
        task_2,  # The one ti if task_2 is not mapped, or all tis if mapped.
    ],
)
```

That would involve a larger change though, or at least that’s my impression when I last looked into this (before I got distracted by other issues).

Reply:
Updated